BUILDING.md

```diff
@@ -45,6 +45,21 @@ Check that `cargo` is correctly installed by running
 cargo --version
 ```
 
+Next, head to https://rust-lang.github.io/rustup-components-history/ and find the most recent date that all the Tier1 platforms have a present rustfmt. For example: nightly-2019-03-04.
+
+Next, install that nightly version of rustfmt
+```
+rustup toolchain install nightly nightly-2019-03-04
+rustup component add rustfmt --toolchain nightly-2019-03-04 rustfmt
+```
```

Comment:
I think @baumanj's suggestion (https://github.com/habitat-sh/habitat/pull/6570#discussion_r284925996) was, rather than hard-code a specific nightly version in this example, directly source the contents of the file that we keep updated with the latest nightly version that works.

That way, you can just run that command, and it will get the version we're currently using.

Reply:
☝️

Reply:
Yeah, I think we should suggest people to just run
```bash
cargo +"$(< RUSTFMT_VERSION)" fmt
```

Reply:
Won't that only work at the root level of the Habitat project, though? I tried it from within a component directory and it did not work.

Reply:
This is a command you'd want to run from the root. If run from a component directory, I think the
```
bash: RUSTFMT_VERSION: No such file or directory
```
error should be sufficient to let users know what's gone wrong.

Reply:
I'd much rather err on the side of clarity and be explicit.